src/utils.ts

```diff
@@ -717,6 +712,16 @@ export function maxWireVersion(topologyOrServer?: Connection | Topology | Server
   return 0;
 }
 
+export function getTopology(provider: MongoClient | Db | Collection): Topology | undefined {
+  const Collection = loadCollection();
+  const Db = loadDb();
+  const MongoClient = loadMongoClient();
+
+  if (provider instanceof MongoClient) return provider.topology;
+  if (provider instanceof Db) return provider.s.client.topology;
+  if (provider instanceof Collection) return provider.s.db.s.client.topology;
+}
+
 /**
```

Comment:
I included the dynamic loaders because I was having circular dependency issues once I added this function. Now, I'm running into issues because the linter thinks the `MongoClient` import is unused (I am assuming because it seems to be overrided here). 

I can see at least two approaches: one is to change the argument type to `OperationParent`, another is to move this function to the `topology.ts` file, where the circular dependency issue is not happening. Is there one of these, or another approach, that is preferred?